pkg/discovery/dtofactory/cluster_dto_builder.go

```diff
@@ -14,29 +14,37 @@ import (
 	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
 )
 
-const ClusterKeyPrefix = "cluster-"
+const (
+	ClusterKeyPrefix = "cluster-"
+	ClusterFreeKey   = "ClusterFreeKey"
+)
 
 var quotaToComputeResourceCommMap = map[proto.CommodityDTO_CommodityType]proto.CommodityDTO_CommodityType{
 	proto.CommodityDTO_VCPU_LIMIT_QUOTA: proto.CommodityDTO_VCPU,
 	proto.CommodityDTO_VMEM_LIMIT_QUOTA: proto.CommodityDTO_VMEM,
 }
 
 type clusterDTOBuilder struct {
-	cluster  *repository.ClusterSummary
-	targetId string
+	cluster               *repository.ClusterSummary
+	targetId              string
+	enablePodClusterMoves bool
 }
 
 func NewClusterDTOBuilder(cluster *repository.ClusterSummary,
-	targetId string) *clusterDTOBuilder {
+	targetId string, enablePodClusterMoves bool) *clusterDTOBuilder {
 	return &clusterDTOBuilder{
-		cluster:  cluster,
-		targetId: targetId,
+		cluster:               cluster,
+		targetId:              targetId,
+		enablePodClusterMoves: enablePodClusterMoves,
 	}
 }
 
 // GetClusterKey constructs the commodity key sold by the cluster entity, by adding a prefix to the cluster id.
 // Use of a prefix is to distinguish the same key used by the node entities in the cluster
-func GetClusterKey(clusterId string) string {
+func GetClusterKey(clusterId string, enablePodClusterMoves bool) string {
+	if enablePodClusterMoves {
+		return ClusterFreeKey
+	}
```

Comment:
I'm not sure if we should allow customization of the key between cluster and namespace entities.  Let's discuss.  I think we want such a customization between pods and nodes/VMs (so pods can move across VMs in different clusters).

Reply:
Code modified based on our discussion. Thanks!